packages/core/src/typeschema/types.ts

```diff
@@ -138,7 +146,26 @@ export function loadDataType(sd: StructureDefinition, profileUrl?: string | unde
 
   const dataTypes = getDataTypesMap(profileUrl);
 
-  dataTypes[sd.name] = schema;
+  // By default, only index by "type" for "official" FHIR types
+  // Note that if profileUrl is provided, then dataTypes will be PROFILE_DATA_TYPES[profileUrl].
+  if (
+    sd.url === `http://hl7.org/fhir/StructureDefinition/${sd.type}` ||
+    sd.url === `https://medplum.com/fhir/StructureDefinition/${sd.type}` ||
+    sd.type?.startsWith('http://') ||
+    sd.type?.startsWith('https://') ||
+    profileUrl
+  ) {
+    dataTypes[sd.type] = schema;
+  }
+
+  // Special cases by "name"
+  // These are StructureDefinitions that are technically "profiles", but are used as base types
+  if (!profileUrl) {
+    const specialCase = TYPE_SPECIAL_CASES[sd.url];
+    if (specialCase) {
+      dataTypes[specialCase] = schema;
+    }
+  }
```

Comment:
Does the same logic changes looking for special cases and using type vs name need to be applied in the `for (const inner of schema.innerTypes)` loop down below?

Reply:
Fixed ✅